### PR TITLE
Refactor AuthServlet

### PR DESCRIPTION
**Pull request by edgarzhavoronkov**
Added method that accepts function on access token from provider and made GoogleDriveInfoGetter use it instead of ripping apart AuthServlet

### CODE REVIEW

File: src/com/bardsoftware/server/auth/AuthServlet.java [hunk 1]
```diff
@@ -153,8 +136,36 @@ private String doOauth(HttpApi http, String authProvider, DefaultOAuthPlugin plu
     return doOauthWithCallback(http, plugin, myUrlService.getUrl("oauth.callback", http) + authProvider);
   }
 
+  private String doDevAuth(HttpApi http, DefaultOAuthPlugin plugin) throws IOException {
```
Comment (dbarashev) [inline src/com/bardsoftware/server/auth/AuthServlet.java]: Can you move it to line 216 where it used to be? Hopefully the diff will look better

File: src/com/bardsoftware/server/auth/AuthServlet.java [hunk 2]
```diff
@@ -153,8 +136,36 @@ private String doOauth(HttpApi http, String authProvider, DefaultOAuthPlugin plu
...
+    String id = http.getUrlParameter("id");
+    String name = http.getUrlParameter("name");
+    if (Strings.isNullOrEmpty(id) || Strings.isNullOrEmpty(name)) {
+      http.sendRedirect(myUrlService.buildUrlFromPath(plugin.getProperty(".redirect"), http));
+      return null;
+    }
+    return MessageFormat.format("'{'\"id\" : \"{0}\", \"name\" : \"{1}\"'}'", id, name);
+  }
+
+  protected DefaultOAuthPlugin getOauthPlugin(String authProvider) {
```
Comment (dbarashev) [inline src/com/bardsoftware/server/auth/AuthServlet.java]: And these two back to where they used to be, above doGet

File: src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  public GoogleDriveInfoGetter(PrincipalExtent principalExtent, AppCapabilitiesService capabilities, AppUrlService urlService, Properties properties) {
+    super(principalExtent, capabilities, urlService, properties);
+  }
+
+  // Basically we need access token and given name
+  public JSONObject getGoogleDriveInfo(HttpApi http, String callback) throws IOException {
+    try {
+      Properties props = getProperties();
+      final String googleDriveScope = props.getProperty("google.drive.scope");
```
Comment (dbarashev) [inline src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java]: Perhaps for Google Drive we can just hardcode the scope itself. Fetching from prpoerties makes sense when you may have different scope names fro different services, but Google Drive is the only one.

File: src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          .provider(plugin.getBuilderApiClass())
+          .apiKey(plugin.getKey())
+          .apiSecret(plugin.getSecret())
+          .callback(callback)
+          .scope(googleDriveScope);
+
+      OAuthService service = serviceBuilder.build();
+
+      return doOauthWithCallbackAndTokenHandler(http, plugin, service, token -> {
+        OAuthRequest givenNameRequest = new OAuthRequest(Verb.GET, plugin.buildRequest(token.getRawResponse()));
```
Comment (dbarashev) [inline src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java]: I understand that the essential part is granting scope access, while this request is not that necessary, but it is not quiet obvious from the code. Can you add a comment, like "we need to show somehow that Drive is connected successfully, let's use user name for that"?

File: src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java (hunk 3 above)
Comment (edgarzhavoronkov) [reply src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java]: Done

File: src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java (hunk 4 above)
Comment (edgarzhavoronkov) [reply src/com/bardsoftware/server/auth/GoogleDriveInfoGetter.java]: Done

**Review by dbarashev — APPROVED** [review]
LGTM

**Pull request merged**